compiler/resolution/virtualDispatch.cpp

```diff
@@ -271,6 +286,19 @@ static void collectMethods(FnSymbol*               pfn,
         }
         if (possibleSignatureMatch(pfn, cfn) == true) {
           methods.push_back(cfn);
+
+          // check to see if we are using defaulted actual fns
+          // for any of the formals in pfn. If so, make sure the
+          // corresponding ones exist for the child.
+          int nUserFormals = getNumUserFormals(pfn);
+          for (int i = 1; i <= nUserFormals; i++) {
+            ArgSymbol* pformal = getUserFormal(pfn, i);
+            ArgSymbol* cformal = getUserFormal(cfn, i);
+            FnSymbol* pDefFn = findExistingDefaultedActualFn(pfn, pformal);
+            if (pDefFn) {
+              getOrCreateDefaultedActualFn(cfn, cformal);
+            }
+          }
```

Comment:
I would prefer to iterate directly over the formals' DefExpr via their `next` links, to avoid quadratic time for this loop. For now, OK to keep as is.

Reply:
I'll leave it alone for now because there are many other loops in the file that follow the same pattern and I don't want to get into updating them all in this PR. Besides that, it seems that having O(1) access to the i'th formal might be the better way to solve the problem.